src/weapon/Weapon.ts

```diff
@@ -0,0 +1,198 @@
+import { WeaponComponentCollection } from '../weaponComponent/WeaponComponentCollection';
+import { Ped } from '../models';
+import { AmmoType, WeaponHash } from '../hashes';
+import { WeaponDisplayNameByHash } from './WeaponDisplayNameByHash';
+import { Game } from '../Game';
+import { Model } from '../Model';
+import { WeaponTint } from './WeaponTint';
+import { WeaponGroup } from './WeaponGroup';
+import { WeaponLivery } from './WeaponLivery';
+import { WeaponLiveryColor } from './WeaponLiveryColor';
+import { WeaponHudStats } from './WeaponHudStats';
+import { enumValues } from '../utils/EnumValues';
+import { Mk2WeaponHash } from './Mk2WeaponHash';
+
+export class Weapon {
+  private readonly owner: Ped;
+  private readonly components: WeaponComponentCollection;
+  private readonly hash: WeaponHash;
+
+  constructor(owner?: Ped, hash?: WeaponHash) {
+    if (owner && hash) {
+      this.owner = owner;
+      this.hash = hash;
+      this.components = new WeaponComponentCollection(this.owner, this);
+    } else {
+      this.hash = WeaponHash.Unarmed;
+    }
+  }
+
+  // public get Components(): WeaponComponentCollection {
```

Comment:
Since you initialize `this.components` in the constructor now, we can remove this comment out code?

Reply:
This leftover will be removed.